### PR TITLE
[cleanup] Upgrade more doc following the PyCQA migration

### PR DESCRIPTION
**Pull request by Pierre-Sassoulas**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :scroll: Docs          |

## Description

Replace #8515

### CODE REVIEW

**Review by DanielNoord — APPROVED** [review]

File: script/check_newsfragments.py [hunk 1]
```diff
@@ -39,7 +39,7 @@
 ISSUES_KEYWORDS = "|".join(VALID_ISSUES_KEYWORDS)
 VALID_CHANGELOG_PATTERN = (
     rf"(?P<description>(.*\n)*(.*\.\n))\n(?P<ref>({ISSUES_KEYWORDS})"
-    r" (PyCQA/astroid)?#(?P<issue>\d+))"
+    r" (pylint-dev/astroid)?#(?P<issue>\d+))"
```
Comment (Pierre-Sassoulas) [inline script/check_newsfragments.py:42]: Huho, this is a regex
```suggestion
    r" (pylint\-dev/astroid)?#(?P<issue>\d+))"
```

**Review by DanielNoord — APPROVED** [review]

**Comment by Pierre-Sassoulas** [comment]
@DanielNoord any ideas why the cache version change did not work for primers ?

**Comment by DanielNoord** [comment]
@Pierre-Sassoulas I haven't checked, but isn't the cache only built on `main` and only retrieved on branches? That would mean that the cache bump needs to go to `main` first.

**Comment by Pierre-Sassoulas** [comment]
That's what I hoped :D

File: doc/whatsnew/2/2.13/summary.rst [hunk 2]
```diff
@@ -295,13 +295,13 @@ Other Changes
...
   Closes #5408
-  RefsPyCQA/astroid#1392
+  Refspylint-dev/astroid#1392
 
 * Fixed false positive for ``unused-argument`` when a method overridden in a subclass
   does nothing with the value of a keyword-only argument.
 
   Closes #5771
-  RefsPyCQA/astroid#1382
+  Refspylint-dev/astroid#1382
```
Comment (DanielNoord) [inline doc/whatsnew/2/2.13/summary.rst:304]: This looks very weird, but not for this PR. (Any other for that matter)

**Review by DanielNoord — APPROVED** [review]

File: doc/whatsnew/2/2.13/summary.rst (hunk 2 above)
Comment (Pierre-Sassoulas) [reply doc/whatsnew/2/2.13/summary.rst:304]: https://github.com/pylint-dev/pylint/pull/8519 it was a sed away, so I did it.

**Pull request merged**